RecoVertex/PrimaryVertexProducer/src/DAClusterizerInZT.cc

```diff
@@ -0,0 +1,778 @@
+#include "RecoVertex/PrimaryVertexProducer/interface/DAClusterizerInZT.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+#include "DataFormats/GeometryCommonDetAlgo/interface/Measurement1D.h"
+#include "RecoVertex/VertexPrimitives/interface/VertexException.h"
+#include "FWCore/Utilities/interface/isFinite.h"
+
+using namespace std;
+
+
+namespace {
+  constexpr double dtCutOff = 4.0;
+
+  bool recTrackLessZ1(const DAClusterizerInZT::track_t & tk1,
+                      const DAClusterizerInZT::track_t & tk2)
+  {
+    return tk1.z < tk2.z;
+  }
+}
+
+
+vector<DAClusterizerInZT::track_t> 
+DAClusterizerInZT::fill( const vector<reco::TransientTrack> & tracks )const{
+  // prepare track data for clustering
+  vector<track_t> tks;
+  for(vector<reco::TransientTrack>::const_iterator it=tracks.begin(); it!=tracks.end(); it++){
+    track_t t;
+    t.pi = 1.;
+    t.z = ((*it).stateAtBeamLine().trackStateAtPCA()).position().z();
+    t.t = it->timeExt(); // the time
+    
+    if (std::abs(t.z) > 1000.) continue;
+    auto const & t_mom = (*it).stateAtBeamLine().trackStateAtPCA().momentum();
+    //  get the beam-spot
+    reco::BeamSpot beamspot = (it->stateAtBeamLine()).beamSpot();
+    double t_dz2 = 
+      std::pow((*it).track().dzError(), 2) // track errror
+      + (std::pow(beamspot.BeamWidthX()*t_mom.x(),2)+std::pow(beamspot.BeamWidthY()*t_mom.y(),2))*std::pow(t_mom.z(),2)/std::pow(t_mom.perp2(),2) // beam spot width
+      + std::pow(vertexSize_, 2); // intrinsic vertex size, safer for outliers and short lived decays
+    t.dz2 = 1./ t_dz2;
+    
+    t.dtz = 0.;
+    t.dt2 = std::pow((*it).dtErrorExt(),2.0) + std::pow(0.010,2.0);   // the ~injected~ timing error, need to add a small minimum vertex size in time
```

Comment:
here and the rest of the file, consider define ```double sqr(double f){return f*f;}``` in the anon namespace above and use it here, instead of all the std::pow.

0.010^2 should be replaced by 1E-4